annotator-core/src/main/java/edu/ucr/cs/riple/core/metadata/trackers/generatedcode/LombokTracker.java

```diff
@@ -0,0 +1,77 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.core.metadata.trackers.generatedcode;
+
+import edu.ucr.cs.riple.core.metadata.method.MethodDeclarationTree;
+import edu.ucr.cs.riple.core.metadata.trackers.MethodRegionTracker;
+import edu.ucr.cs.riple.core.metadata.trackers.Region;
+import edu.ucr.cs.riple.scanner.generatedcode.SourceType;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Tracker for extending potentially impacted regions for elements which will be use in generated
+ * code by <a href="https://projectlombok.org">Lombok</a>. Lombok automatically propagates
+ * {@code @Nullable} annotation on fields to getter methods, therefore, extends the set of
+ * potentially impacted regions to all callers of that method as well. This tracker, will include
+ * all callers of any method which uses an element in a lombok generated code. This will guarantee
+ * that {@link edu.ucr.cs.riple.core.evaluators.graphprocessor.ParallelConflictGraphProcessor} will
+ * catch any triggered errors by an annotation including all copied annotations by lombok as well.
+ */
+public class LombokTracker implements GeneratedRegionTracker {
+
+  /** Method region tracker to get potentially impacted regions of a method. */
+  private final MethodRegionTracker tracker;
+  /** Method declaration tree instance. */
+  private final MethodDeclarationTree methodDeclarationTree;
+
+  public LombokTracker(
+      MethodDeclarationTree methodDeclarationTree, MethodRegionTracker methodRegionTracker) {
+    this.methodDeclarationTree = methodDeclarationTree;
+    this.tracker = methodRegionTracker;
+  }
+
+  @Override
+  public Set<Region> extendWithGeneratedRegions(Set<Region> regions) {
+    return regions.stream()
+        // filter regions which are created by lombok
+        .filter(region -> region.sourceType.equals(SourceType.LOMBOK) && region.isOnMethod())
+        // find the corresponding method for the region.
+        .map(region -> methodDeclarationTree.findNode(region.member, region.clazz))
+        .filter(Objects::nonNull)
+        // get method location.
+        .map(methodNode -> methodNode.location)
+        // add potentially impacted regions for the collected methods.
+        .flatMap(
+            onMethod -> {
+              Optional<Set<Region>> ans = tracker.getRegions(onMethod);
```

Comment:
Not related to this PR, but it seems strange that the `getRegions()` method returns an `Optional<Set<Region>>` rather than just a `Set<Region>` that could be empty.  Maybe fix in a follow up

Reply:
@msridhar I wanted to distinguish between trackers "not having any knowledge about the passed element" and "having data but not having any actual usage". But this was required in the old designs and should be updated in follow up PRs.